src/plots/cartesian/axes.js

```diff
@@ -620,6 +619,16 @@ axes.calcTicks = function calcTicks(ax, opts) {
 
     generateTicks();
 
+    var isPeriod = ax.ticklabelmode === 'period';
+    var addOneTickToStart = isPeriod && tickVals.length > 2;
+
+    if(addOneTickToStart) {
+        tickVals.push({
+            minor: false,
+            value: 2 * tickVals[0].value - tickVals[1].value
+        });
```

Comment:
Looking good! Maybe this is what you were referring to this morning, but I see a little funny behavior with the first label: when the first tick is present, the first label is correct, but when the first tick is missing, the first label is sometimes incorrect. I suspect this can be fixed by using `tickIncrement` and inverting the 3rd arg, something like:
`value: axes.tickIncrement(tickVals[0].value, ax.dtick, !axrev, ax.caldendar)`

Should be 1886, second one is correct:
<img width="481" alt="Screen Shot 2020-07-23 at 4 06 10 PM" src="https://user-images.githubusercontent.com/2678795/88334016-1d635f80-ccff-11ea-92be-3dac5a82d506.png">
<img width="509" alt="Screen Shot 2020-07-23 at 4 06 39 PM" src="https://user-images.githubusercontent.com/2678795/88334024-22c0aa00-ccff-11ea-982c-b76fe1c70a33.png">

Should be 1887, second one is correct:
<img width="490" alt="Screen Shot 2020-07-23 at 4 08 55 PM" src="https://user-images.githubusercontent.com/2678795/88334050-2c4a1200-ccff-11ea-89a0-cbb41d72dc5a.png">
<img width="476" alt="Screen Shot 2020-07-23 at 4 09 07 PM" src="https://user-images.githubusercontent.com/2678795/88334085-34a24d00-ccff-11ea-9aff-c16ecc2c10bf.png">

And with months, May is correct:
<img width="480" alt="Screen Shot 2020-07-23 at 4 14 15 PM" src="https://user-images.githubusercontent.com/2678795/88334373-a7132d00-ccff-11ea-9302-22e415dcee10.png">
<img width="467" alt="Screen Shot 2020-07-23 at 4 14 28 PM" src="https://user-images.githubusercontent.com/2678795/88334385-aaa6b400-ccff-11ea-9c93-74a9ea4d7533.png">

Day tick labels look to have the correct values, but I notice a first one with no tick doesn't get the year, it goes on the original first tick, which is now the second one. Maybe this is as simple as putting this tick first rather than last in the array? ie `unshift` rather than `push`?
<img width="470" alt="Screen Shot 2020-07-23 at 4 16 47 PM" src="https://user-images.githubusercontent.com/2678795/88334655-1ab53a00-cd00-11ea-9c49-686b42d4f5c5.png">

Reply:
Thanks for the review @alexcjohnson 
Let me give this part a second try / tries.

Reply:
@alexcjohnson I applied your suggestions in https://github.com/plotly/plotly.js/pull/4993/commits/79385c24e5fe80aff8008f8818a11582146031a4 and https://github.com/plotly/plotly.js/pull/4993/commits/66a98be64f76ecfb43ed04da72500251f4bbe694.